crates/turbo-trace/src/tracer.rs

```diff
@@ -79,104 +237,104 @@ impl Tracer {
             });
         }
 
-        let resolver = Resolver::new(options);
-        let mut errors = vec![];
+        Resolver::new(options)
+    }
+
+    pub async fn trace(mut self, max_depth: Option<usize>) -> TraceResult {
         let mut seen: HashMap<AbsoluteSystemPathBuf, SeenFile> = HashMap::new();
+        let resolver = self.create_resolver();
 
         while let Some((file_path, file_depth)) = self.files.pop() {
-            if matches!(file_path.extension(), Some("json") | Some("css")) {
-                continue;
-            }
-
-            if seen.contains_key(&file_path) {
-                continue;
-            }
-
             if let Some(max_depth) = max_depth {
                 if file_depth > max_depth {
                     continue;
                 }
             }
+            self.trace_file(&resolver, file_path, file_depth, &mut seen)
+                .await;
+        }
 
-            let entry = seen.entry(file_path.clone()).or_default();
-
-            // Read the file content
-            let Ok(file_content) = fs::read_to_string(&file_path) else {
-                errors.push(TraceError::FileNotFound(file_path.clone()));
-                continue;
-            };
-
-            let comments = SingleThreadedComments::default();
-
-            let source_file = self.source_map.new_source_file(
-                FileName::Custom(file_path.to_string()).into(),
-                file_content.clone(),
-            );
-
-            let syntax =
-                if file_path.extension() == Some("ts") || file_path.extension() == Some("tsx") {
-                    Syntax::Typescript(TsSyntax {
-                        tsx: file_path.extension() == Some("tsx"),
-                        decorators: true,
-                        ..Default::default()
-                    })
-                } else {
-                    Syntax::Es(EsSyntax {
-                        jsx: file_path.ends_with(".jsx"),
-                        ..Default::default()
-                    })
-                };
-
-            let lexer = Lexer::new(
-                syntax,
-                EsVersion::EsNext,
-                StringInput::from(&*source_file),
-                Some(&comments),
-            );
+        TraceResult {
+            files: seen,
+            errors: self.errors,
+        }
+    }
 
-            let mut parser = Parser::new_from(Capturing::new(lexer));
+    pub async fn reverse_trace(mut self) -> TraceResult {
+        let files = match globwalk::globwalk(
```

Comment:
I'd love at some point to combine the glob walking and the import resolution so that we can have more concurrency. Right now we have to walk the entire repo, then do the import resolution step.